### PR TITLE
Fix failing tests

### PR DESCRIPTION
**Pull request by rsokl**
torch's softmax crossentropy now requires long-type labels

### CODE REVIEW

File: tests/test_boxes.py [hunk 1]
```diff
@@ -115,37 +140,57 @@ def test_shapes(self, boxes: ndarray, truth: ndarray, data: st.SearchStrategy):
         K = truth.shape[0]
         labels = data.draw(hnp.arrays(dtype=int, shape=(K,)))
         cls, reg = generate_targets(boxes, truth, labels, 0.5, 0.4)
-        assert cls.shape == (N,), 'generate_targets failed to produce classification targets of the correct shape'
-        assert reg.shape == (N, 4), 'generate_targets failed to produce regression targets of the correct shape'
+        assert cls.shape == (
+            N,
+        ), "generate_targets failed to produce classification targets of the correct shape"
```
Comment (davidmascharka) [inline tests/test_boxes.py]: this looks terrible....I'd rather change this to

``` python
msg = "generate_targets failed to produce classification targets of the correct shape"
assert cls.shape == (N,), msg

msg = "generate_targets failed to produce regression targets of the correct shape"
assert reg.shape == (N, 4), msg
```

or something

File: tests/test_boxes.py [hunk 2]
```diff
@@ -115,37 +140,57 @@ def test_shapes(self, boxes: ndarray, truth: ndarray, data: st.SearchStrategy):
...
+        assert reg.shape == (
+            N,
+            4,
+        ), "generate_targets failed to produce regression targets of the correct shape"
```
Comment (davidmascharka) [inline tests/test_boxes.py]: see above

File: tests/test_boxes.py [hunk 3]
```diff
@@ -115,37 +140,57 @@ def test_shapes(self, boxes: ndarray, truth: ndarray, data: st.SearchStrategy):
...
-        prop = np.array([[-0.5, -0.5, 0.5, 0.5],      # neither axis matches truth
-                         [0, -0.5, np.exp(1), 0.5],   # x matches truth
-                         [-0.5, 0, 0.5, np.exp(1)]])  # y matches truth
+        prop = np.array(
+            [
+                [-0.5, -0.5, 0.5, 0.5],  # neither axis matches truth
+                [0, -0.5, np.exp(1), 0.5],  # x matches truth
+                [-0.5, 0, 0.5, np.exp(1)],
+            ]
+        )  # y matches truth
```
Comment (davidmascharka) [inline tests/test_boxes.py]: this comment should be associated with the line `[-0.5, 0, 0.5, np.exp(1)],`

File: tests/test_boxes.py [hunk 4]
```diff
@@ -115,37 +140,57 @@ def test_shapes(self, boxes: ndarray, truth: ndarray, data: st.SearchStrategy):
...
         truth = np.array([[0, 0, np.exp(1), np.exp(1)]])
         labels = np.array([1.0])
 
-        out_reg = np.array([[np.exp(1) / 2, np.exp(1) / 2, 1, 1],
-                            [0, np.exp(1) / 2, 0, 1],
-                            [np.exp(1) / 2, 0, 1, 0]])
+        out_reg = np.array(
+            [
```
Comment (davidmascharka) [inline tests/test_boxes.py:176]: is black not happy with having `([` on a single line?

File: tests/test_boxes.py [hunk 5]
```diff
@@ -155,24 +200,31 @@ def test_label_invariance(self, label0: int, label1: int, shuffle_inds: List[int
                          [10., 10., 11, 11]])   # iou = 1 (truth 1) should be marked positive
 
         # xyxy format
-        truth = np.array([[-.5, -.5, .5, .5],
-                          [10., 10., 11, 11]])
+        truth = np.array([[-0.5, -0.5, 0.5, 0.5], [10.0, 10.0, 11, 11]])
```
Comment (davidmascharka) [inline tests/test_boxes.py]: I prefer this on two lines. Will black leave it alone if we just change the indent?

``` python
truth = np.array([
    [-0.5, -0.5, 0.5, 0.5],
    [10.0, 10.0, 11, 11],
])
```

File: tests/test_boxes.py [hunk 6]
```diff
@@ -204,32 +260,30 @@ def test_identical(self, x, score, rep):
...
+        msg = "non_max_suppression failed to produce the expected output for a single detection"
         assert_array_equal(nms, np.array([0]), msg)
 
     @pytest.mark.parametrize(
         ("threshold", "desired_nms"),
-        [(0.5, np.array([0, 1])),
-         (0.25, np.array([0, 1])),
-         (0.15, np.array([1])),
-         ]
+        [(0.5, np.array([0, 1])), (0.25, np.array([0, 1])), (0.15, np.array([1]))],
```
Comment (davidmascharka) [inline tests/test_boxes.py]: I liked this better before.....can we enforce that?

File: tests/test_boxes.py [hunk 7]
```diff
@@ -204,32 +260,30 @@ def test_identical(self, x, score, rep):
...
     )
     def test_known_results(self, threshold, desired_nms):
         """ Ensures that non_max_suppression works correctly for known values. """
-        boxes = np.array([[0, 0, 1, 1],
-                          [0.5, 0.5, 0.9, 0.9]])
+        boxes = np.array([[0, 0, 1, 1], [0.5, 0.5, 0.9, 0.9]])
```
Comment (davidmascharka) [inline tests/test_boxes.py]: same... can this be

``` python
boxes = np.array([
    [0, 0, 1, 1],
    [0.5, 0.5, 0.9, 0.9],
])
```

File: tests/test_metrics.py [hunk 8]
```diff
@@ -29,36 +29,71 @@ class Test_precision_recall:
...
+        [
+            (detections_a, detections_a, 1.0, "identical boxes"),
+            (detections_a, detections_b, 0.0, "identical box with different class"),
+            (detections_a, detections_c, 0.0, "non-overlapping boxes"),
+            (
+                detections_b,
+                detections_c,
+                0.0,
+                "non-overlapping boxes with different classes",
+            ),
```
Comment (davidmascharka) [inline tests/test_metrics.py]: this is awful....it completely breaks the flow -- are there alternatives/can we just have black respect a slightly longer line length (like 100?)

File: tests/test_metrics.py [hunk 9]
```diff
@@ -29,36 +29,71 @@ class Test_precision_recall:
...
```
Comment (davidmascharka) [inline tests/test_metrics.py]: same here

**Review by davidmascharka — COMMENTED** [review]
Looks great overall! Just a couple minor things I'm not sure we can change

**Comment by rsokl** [comment]
@davidmascharka I agree with all of your feedback. I undid a lot of the auto-formatting. I think this should be good to go

**Review by davidmascharka — APPROVED** [review]
Looks good. Thanks for the fix(?) :thinking:
